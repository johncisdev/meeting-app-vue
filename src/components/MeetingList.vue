<template>
  <v-container>
    <v-row>
      <v-col cols="12">
        <v-row>
          <v-col cols="6"> <div class="text-h6 mb-3">Meetings</div></v-col>

          <v-col cols="6" class="text-right">
            <v-btn
              color="primary"
              elevation="2"
              raised
              small
              @click="openFormDialog"
              >Create Meeting</v-btn
            ></v-col
          >
        </v-row>
        <v-card-title>
          <v-text-field
            v-model="search"
            append-icon="mdi-magnify"
            label="Search"
            single-line
            hide-details
          ></v-text-field>
        </v-card-title>
        <v-data-table
          :headers="headers"
          :items="data"
          :items-per-page="10"
          :search="search"
          class="elevation-1"
        >
          <template v-slot:item.hostName="{ item }">
            <v-avatar
              class="accent white--text mr-1"
              size="25"
              v-text="item.hostName.slice(0, 1).toUpperCase()"
            >
            </v-avatar>
            {{ item.hostName }}
          </template>
          <template v-slot:item.participants="{ item }">
            <v-avatar
              class="accent white--text"
              size="25"
              v-for="participant in item.participants"
              :key="participant"
              v-text="participant.charAt(0).toUpperCase()"
            ></v-avatar>
            {{ participant }}
          </template>
          <template v-slot:item.time="{ item }">
            {{ item.bookingStartTime }} - {{ item.bookingEndTime }}
          </template>
          <template v-slot:item.action="{ item }">
            <v-menu bottom left>
              <template v-slot:activator="{ on, attrs }">
                <v-btn dark color="primary" icon v-bind="attrs" v-on="on">
                  <v-icon>mdi-dots-vertical</v-icon>
                </v-btn>
              </template>

              <v-list>
                <v-list-item>
                  <v-list-item-title
                    @click="
                      $router.push({
                        path: `/meeting/${item.id}`,
                      })
                    "
                    >View Meeting</v-list-item-title
                  >
                </v-list-item>
                <v-list-item>
                  <v-list-item-title @click="editMeeting(item)"
                    >Edit</v-list-item-title
                  >
                </v-list-item>
                <v-list-item>
                  <v-list-item-title @click="deleteMeeting(item)"
                    >Delete</v-list-item-title
                  >
                </v-list-item>
              </v-list>
            </v-menu>
          </template>
        </v-data-table>
        <v-snackbar v-model="snackbar">
          {{ message }}
          <template v-slot:action="{ attrs }">
            <v-btn color="blue" text v-bind="attrs" @click="snackbar = false">
              Close
            </v-btn>
          </template>
        </v-snackbar>
      </v-col>
    </v-row>
    <meeting-form
      v-show="isOpenFormDialog"
      :is-open="isOpenFormDialog"
      :selected="selected"
      @openFormDialog="openFormDialog"
      @showSnackbar="showSnackbar"
    />
  </v-container>
</template>

<script>
import MeetingForm from "./MeetingForm";

export default {
  name: "MeetingList",
  components: {
    MeetingForm,
  },
  data() {
    return {
      search: "",
      headers: [
        {
          text: "Room Name",
          value: "name",
        },
        { text: "Host", value: "hostName" },
        { text: "Participants", value: "participants" },
        { text: "Date", value: "bookingDate" },
        { text: "Time", value: "time" },
        { text: "", value: "action" },
      ],
      isOpenFormDialog: false,
      selected: {},
      message: "",
      snackbar: false,
    };
  },
  computed: {
    data() {
      return this.$store.state.dataList;
    },
  },
  async created() {
    await this.$store.dispatch("getMeetings");
  },
  methods: {
    openFormDialog() {
      this.isOpenFormDialog = !this.isOpenFormDialog;
    },
    showSnackbar(message) {
      this.message = message;
      this.snackbar = true;
    },
    editMeeting(meeting) {
      this.selected = meeting;
      this.isOpenFormDialog = true;
    },
    deleteMeeting(meeting) {
      this.$store.dispatch("deleteMeeting", meeting.id);
      this.showSnackbar("Meeting was successfully deleted.");
    },
  },
};
</script>
