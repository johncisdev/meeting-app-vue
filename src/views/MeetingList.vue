<template>
  <meeting-list />
</template>

<script>
import MeetingList from "../components/MeetingList";

export default {
  name: "Meetings",

  components: {
    MeetingList,
  },
};
</script>
