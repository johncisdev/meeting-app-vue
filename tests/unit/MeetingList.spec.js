import { shallowMount, createLocalVue, mount } from "@vue/test-utils";
import Vuex from "vuex";

import MeetingList from "@/components/MeetingList.vue";
import Store from "@/store/index";

describe("MeetingList.vue test suite", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let store;

  beforeEach(() => {
    store = new Vuex.Store(Store);
  });

  it("should match with the snapshot", () => {
    const wrapper = mount(MeetingList, {
      store,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render the component", () => {
    const wrapper = shallowMount(MeetingList, {
      store,
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("should have add button", () => {
    const wrapper = shallowMount(MeetingList, {
      store,
    });
    expect(wrapper.find("v-btn").exists()).toBe(true);
    expect(wrapper.find("v-btn").text()).toBe("Create Meeting");
  });

  it("should have a global search", () => {
    const wrapper = shallowMount(MeetingList, {
      store,
    });
    expect(wrapper.find("v-text-field").exists()).toBe(true);
  });

  it("should have a table", () => {
    const wrapper = shallowMount(MeetingList, {
      store,
    });
    expect(wrapper.find("v-data-table").exists()).toBe(true);
  });
});
