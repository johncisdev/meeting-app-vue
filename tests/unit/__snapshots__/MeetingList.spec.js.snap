// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingList.vue test suite should match with the snapshot 1`] = `
<v-container>
  <v-row>
    <v-col
      cols="12"
    >
      <v-row>
        <v-col
          cols="6"
        >
          <div
            class="text-h6 mb-3"
          >
            Meetings
          </div>
        </v-col>
         
        <v-col
          class="text-right"
          cols="6"
        >
          <v-btn
            color="primary"
            elevation="2"
            raised=""
            small=""
          >
            Create Meeting
          </v-btn>
        </v-col>
      </v-row>
       
      <v-card-title>
        <v-text-field
          append-icon="mdi-magnify"
          hide-details=""
          label="Search"
          single-line=""
        />
      </v-card-title>
       
      <v-data-table
        class="elevation-1"
        headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        items=""
        items-per-page="10"
        search=""
      />
       
      <v-snackbar>
        
        
        
      </v-snackbar>
    </v-col>
  </v-row>
   
  <v-row
    justify="center"
    style="display: none;"
  >
    <v-dialog
      max-width="600px"
      persistent=""
    >
      <v-card>
        <v-card-title>
          <span
            class="text-h5"
          >
            Create Meeting
          </span>
        </v-card-title>
         
        <v-card-text>
          <v-container>
            <v-row>
              <v-col
                cols="12"
              >
                <v-text-field
                  label="Room Name"
                  rules="value => {
        const error = \\"This Field is required.\\";

        if (value instanceof Array && value.length == 0) {
          return error;
        }

        return !!value || error;
      }"
                />
              </v-col>
               
              <v-col
                cols="12"
              >
                <v-combobox
                  hide-selected=""
                  label="Host Name"
                  persistent-hint=""
                  rules="value => {
        const error = \\"This Field is required.\\";

        if (value instanceof Array && value.length == 0) {
          return error;
        }

        return !!value || error;
      }"
                  small-chips=""
                />
              </v-col>
               
              <v-col
                cols="12"
              >
                <v-combobox
                  hide-selected=""
                  label="Participants"
                  multiple=""
                  persistent-hint=""
                  rules="value => {
        const error = \\"This Field is required.\\";

        if (value instanceof Array && value.length == 0) {
          return error;
        }

        return !!value || error;
      }"
                  small-chips=""
                />
              </v-col>
               
              <v-col
                cols="12"
                sm="4"
              >
                <v-menu
                  max-width="290px"
                  min-width="auto"
                  offset-y=""
                >
                   
                  <v-date-picker
                    no-title=""
                  />
                </v-menu>
              </v-col>
               
              <v-col
                cols="12"
                sm="4"
              >
                <v-menu
                  max-width="290px"
                  min-width="290px"
                  nudge-right="40"
                  offset-y=""
                  transition="scale-transition"
                >
                   
                  <!---->
                </v-menu>
              </v-col>
               
              <v-col
                cols="12"
                sm="4"
              >
                <v-menu
                  max-width="290px"
                  min-width="290px"
                  nudge-right="40"
                  offset-y=""
                  transition="scale-transition"
                >
                   
                  <!---->
                </v-menu>
              </v-col>
               
              <v-col
                cols="12"
                style="display: none;"
              >
                <v-snackbar
                  absolute=""
                  bottom=""
                  color="error"
                  left=""
                  text=""
                  timeout="-1"
                  value="true"
                >
                  
                
              
                </v-snackbar>
              </v-col>
            </v-row>
          </v-container>
        </v-card-text>
         
        <v-card-actions>
          <v-spacer />
           
          <v-btn
            color="blue darken-1"
            text=""
          >
            
          Close
        
          </v-btn>
           
          <v-btn
            color="blue darken-1"
            text=""
          >
             Save 
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</v-container>
`;
